### PR TITLE
Test remote installer

### PR DESCRIPTION
**Pull request by Naatan**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1339" title="DX-1339" target="_blank">TaskDX-1339</a>  We have test plans covering the remote installer
  </td></table>
  <br />
 

Tests surfaced some small issues, which I decided to address while I was at it as otherwise it'd balloon into a timesink to address individually.

### CODE REVIEW

File: cmd/state-remote-installer/main.go [hunk 1]
```diff
@@ -193,12 +196,17 @@ func execute(out output.Outputer, prompt prompt.Prompter, cfg *config.Instance,
 	env := []string{
 		constants.InstallerNoSubshell + "=true",
 	}
-	_, _, err = exeutils.ExecuteAndPipeStd(filepath.Join(tmpDir, constants.StateInstallerCmd+exeutils.Extension), args, env)
+	_, cmd, err := exeutils.ExecuteAndPipeStd(filepath.Join(tmpDir, constants.StateInstallerCmd+exeutils.Extension), args, env)
 	if err != nil {
+		if cmd != nil && cmd.ProcessState.Sys().(syscall.WaitStatus).Exited() {
```
Comment (MDrakos) [inline cmd/state-remote-installer/main.go:201]: Isn't there a possibility of a panic calling `.Exited()` if the type check failed?

File: test/integration/remote_installer_int_test.go [hunk 2]
```diff
@@ -15,23 +17,75 @@ import (
...
+			cp.Expect("Press ENTER to exit")
+			cp.SendLine("")
+			cp.ExpectExitCode(0)
+
+			suite.Require().FileExists(stateExePath)
+
+			cp = ts.SpawnCmdWithOpts(
+				stateExePath,
+				e2e.WithArgs("--version"),
+				e2e.AppendEnv(constants.InstallPathOverrideEnvVarName+"="+installPath),
```
Comment (MDrakos) [inline test/integration/remote_installer_int_test.go:77]: nit: This shouldn't be needed here

**Review by MDrakos — CHANGES_REQUESTED** [review]
Do we want to start phasing out the install scripts integration tests with this PR or wait?

File: cmd/state-remote-installer/main.go (hunk 1 above)
Comment (Naatan) [reply cmd/state-remote-installer/main.go:201]: I thought so too, but this is what Go does:

https://github.com/golang/go/blob/master/src/os/exec_posix.go#L102

File: test/integration/remote_installer_int_test.go (hunk 2 above)
Comment (Naatan) [reply test/integration/remote_installer_int_test.go:77]: It's needed to keep the tests from conflicting with one another. Without this, it always installs in a global directory, which given the asynchronous nature of tests can cause all sorts of issues.

Comment (Naatan) [reply test/integration/remote_installer_int_test.go:77]: Oh you meant it's not needed for `--version`; I don't think that's true either though, as we rely on the install path when calling the svc.

Comment (MDrakos) [reply test/integration/remote_installer_int_test.go:77]: Ahh okay, makes sense 👍

**Review by MDrakos — APPROVED** [review]

**Pull request merged**